src/main/java/com/conveyal/datatools/editor/controllers/api/EditorController.java

```diff
@@ -132,6 +133,49 @@ private void registerRoutes() {
         }
     }
 
+    /**
+     * Endpoint to update multiple trips in a single transaction.
+     * Creation of this endpoint was prompted by the need to allow trip ID "flipping",
+     * where trip IDs may swap values in one request. Previously this case would cause an error
+     * since trips were saved one by one, without knowledge that a target trip ID had been "flipped".
+     */
+    private String updateMultipleTrips(Request req, Response res) {
+        /* TODO: Deleting all trips involved in the update is obviously
+            overkill and will have performance impacts. */
+        // FIXME: how does autoCommit in this method affect performance?
+        String namespace = getNamespaceAndValidateSession(req);
+        try {
+            JsonNode jsonNode = mapper.readTree(req.body());
+            JdbcTableWriter tableWriter = new JdbcTableWriter(table, datasource, namespace);
+            JsonNode trips = jsonNode.get("trips");
+            ArrayList<String> returnedJsonStrings = new ArrayList<>();
+
+            if (jsonNode == null) {
+                logMessageAndHalt(req, 400, "JSON body must be provided with trip update request.");
+            }
+            for (final JsonNode trip: trips) {
+                int id = trip.get("id").intValue();
+                // Id == -2 is the front end indication that the trip is being created.
+                if (id != -2){
+                    tableWriter.delete(id, false);
+                }
+            }
+            for (final JsonNode thisTrip: trips) {
+                String jsonBody = thisTrip.toString();
+                JsonNode lastTrip = trips.get(trips.size()-1);
+                Boolean isLastTrip = lastTrip.equals(thisTrip);
+                // Autocommit: true for last trip
+                returnedJsonStrings.add(tableWriter.create(jsonBody, isLastTrip));
```

Comment:
If trying to "undo" a trip_id change in a row, an error occurs after the following steps (see animation below).

![Screen Recording 2022-05-06 at 1 25 05 PM](https://user-images.githubusercontent.com/56846598/167182718-36c16ff3-0f39-478e-a605-30235604abd9.gif)